src/App.js

```diff
@@ -37,27 +35,24 @@ function App() {
         <main>
           <div className="search-box">
             <input
-              type="text"
               className="search-bar"
-              placeholder="Search..."
               onChange={(e) => setQuery(e.target.value)}
-              value={query}
               onKeyPress={search}
+              placeholder="Search..."
+              type="text"
+              value={query}
             />
           </div>
           <div>
-            {typeof weather.main != 'undefined' ? (
+            {typeof weather.main !== 'undefined' && (
               <div>
-                <Location></Location>
-                <Weather></Weather>
+                <Location weather={weather} />
+                <Weather weather={weather} />
               </div>
-            ) : (
-              ''
             )}
           </div>
         </main>
       </div>
-    </dataContext.Provider>
```

Comment:
Really nice, but we can use the `&&` trick to make this shorter. :) This is because of short-circuit evaluation: If the first statement is not undefined, javascript will not even evaluate the new one. The component "renders" even if it's part of a logical expression, because it has to "return" to know if the value is true or false. As this returns a React object, it will get rendered as objects are `thruthy`.